### PR TITLE
boolean post split fn 

### PR DESCRIPTION
**Pull request by prithvikannan**
## What changes are proposed in this pull request?

Replace `steps.split.process_splits` with `steps.split.mark_to_be_filtered` to ensure users can only filter using this function.

## How is this patch tested?

(Details)

We require adding a link to the [workflow](https://github.com/mlflow/mlflow/actions/workflows/pipeline-template.yml) for testing the [mlflow/mlp-regression-template](https://github.com/mlflow/mlp-regression-template) code path.

Run the workflow with repository name: `mlflow/mlp-regression-template` and the branch`(Example:<user-name>:feature)` that you are working with.
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

https://github.com/mlflow/mlflow/pull/6928

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/regression`: Sklearn regression pipeline example
- [ ] `area/build`: Build and test infrastructure for MLflow pipeline example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: steps/split.py [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
 """
 
-from pandas import DataFrame
+from pandas import DataFrame, Series
 
 
-def process_splits(
-    train_df: DataFrame, validation_df: DataFrame, test_df: DataFrame
-) -> (DataFrame, DataFrame, DataFrame):
+def mark_to_be_filtered(dataset: DataFrame) -> Series(bool):
```
Comment (dbczumar) [inline steps/split.py]: Was the naming of this method agreed upon by stakeholders? Seems a little bit awkward to me. I think `create_dataset_filter` is clearer.

**Review by dbczumar — APPROVED** [review]
LGTM with one comment on method naming.

cc @apurva-koti can we update the full reference docs to accommodate this change?

File: steps/split.py (hunk 1 above)
Comment (apurva-koti) [reply steps/split.py]: +1

Comment (apurva-koti) [reply steps/split.py]: `mark_to_be_filtered_or_not_to_be_filtered_that_is_the_question` gets my vote

Comment (prithvikannan) [reply steps/split.py]: good suggestion! changed to `create_dataset_filter`

**Pull request merged**